### PR TITLE
Feature/add-curriculum-api

### PR DESCRIPTION
**Pull request by freddy5566**
- [x] setUp curriculum api

### CODE REVIEW

File: TAT/Networking/API/APIManager+curriculum.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
+//
+//  APIManager+curriculum.swift
```
Comment (freddy5566) [inline TAT/Networking/API/APIManager+curriculum.swift]: `APIManager+SemesterCurriculumList` would be better

File: TAT/Networking/API/APIManager+curriculum.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+//
+
+import Moya
+import RxSwift
+import RxCocoa
+import SwiftyJSON
+
+extension APIType {
+    
+    struct GetCurriculumList: APITargetType {
```
Comment (freddy5566) [inline TAT/Networking/API/APIManager+curriculum.swift]: as same as file name, SemesterCurriculumList would be better

File: TAT/Networking/API/APIManager+curriculum.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        var task: Task {
+            var params: [String: Any] = [:]
+            params["targetStudentId"] = studentID
+            
+            return .requestParameters(parameters: params,
+                                      encoding: URLEncoding.queryString)
+        }
+    }
+    
+    struct GetCurriculum: APITargetType {
```
Comment (freddy5566) [inline TAT/Networking/API/APIManager+curriculum.swift]: this should split into different file, and file name would be something like `APIManager+YearsCurriculum` and struct name is simply the name after `+`

**Review by yaoandy107 — APPROVED** [review]

File: TAT/Networking/API/APIManager+SemesterCurriculum.swift [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  // MARK: - Unwritten yet
+
+  func fetchSemesterCurriculum(year: String, semester: String, studentID: String?) -> Observable<AnyObject> {
+    let target = MultiTarget(APIType.SemesterCurriculum(year: year, semester: semester, studentID: studentID))
+    return Observable.create { [weak self] (observer) -> Disposable in
+      _ = self?.provider.rx.request(target)
+      .asObservable()
+      .filterSuccessfulStatusCodes()
+      .mapJSON()
+      .parseResponseDirectly(asType: [Course].self)
```
Comment (kamisakihideyoshi) [inline TAT/Networking/API/APIManager+SemesterCurriculum.swift]: The model it parsed would be changed after API updated.

File: TAT/Models/Course.swift [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CourseKeys.self)
+
+    try container.encode(name, forKey: .name)
+    try container.encode(id, forKey: .id)
+    try container.encode(instructor, forKey: .instructor)
+    try container.encode(classroom, forKey: .classroom)
+    try container.encode(periods, forKey: .periods)
+  }
```
Comment (freddy5566) [inline TAT/Models/Course.swift:55]: add one more line at very last.

File: TAT/Models/Periods.swift [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+//
+//  Created by 神崎秀吉 on 2019/02/14.
+//  Copyright © 2019 jamfly. All rights reserved.
+//
+
+import Foundation
+
+class Periods: NSObject, Codable {
+
+    // MARK: - Property
```
Comment (freddy5566) [inline TAT/Models/Periods.swift]: Properties

File: TAT/Models/Periods.swift [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            wednesday = try container.decode([String].self, forKey: .wednesday)
+            thursday = try container.decode([String].self, forKey: .thursday)
+            friday = try container.decode([String].self, forKey: .friday)
+            saturday = try container.decode([String].self, forKey: .saturday)
+        } catch {
+            #if DEBUG
+            print("failed to decode course")
+            #endif
+        }
+    }
```
Comment (freddy5566) [inline TAT/Models/Periods.swift:48]: add one more line at very last.

File: TAT/Models/Periods.swift [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        var container = encoder.container(keyedBy: PeriodKeys.self)
+
+        try container.encode(sunday, forKey: .sunday)
+        try container.encode(monday, forKey: .monday)
+        try container.encode(tuesday, forKey: .tuesday)
+        try container.encode(wednesday, forKey: .wednesday)
+        try container.encode(thursday, forKey: .thursday)
+        try container.encode(friday, forKey: .friday)
+        try container.encode(saturday, forKey: .saturday)
+    }
```
Comment (freddy5566) [inline TAT/Models/Periods.swift:63]: add one more line at very last.

File: TAT/Models/Semester.swift [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+// MARK: - Encodable
+
+extension Semester {
+
+  func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: SemesterKeys.self)
+
+    try container.encode(year, forKey: .year)
+    try container.encode(semester, forKey: .semester)
+  }
```
Comment (freddy5566) [inline TAT/Models/Semester.swift:43]: add one more line at very last.

File: TAT/Networking/API/APIManager+SemesterCurriculum.swift [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+      return "/api/curriculums"
+    }
+
+    var method: Moya.Method {
+      return .get
+    }
+
+    var headers: [String: String]? {
+      guard let token = UserDefaults.standard.string(forKey: "token") else {
+        print("QAQ")
```
Comment (freddy5566) [inline TAT/Networking/API/APIManager+SemesterCurriculum.swift]: print should be inside #if DEBUG, and QAQ should be removed

File: TAT/Networking/API/APIManager+SemesterCurriculum.swift [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+      return .requestParameters(parameters: params,
+                                encoding: URLEncoding.queryString)
+    }
+  }
+
+}
+
+extension APIManager {
+
+  // MARK: - Unwritten yet
```
Comment (freddy5566) [inline TAT/Networking/API/APIManager+SemesterCurriculum.swift]: should be todo or parsing....

File: TAT/Networking/API/APIManager+SemesterCurriculum.swift [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+          print("curriculum is \(courses)")
+          #endif
+        }, onError: { (error) in
+          #if DEBUG
+          print("failed to fetch curriculum with: \(error)")
+          #endif
+        })
+      return Disposables.create()
+    }
+  }
```
Comment (freddy5566) [inline TAT/Networking/API/APIManager+SemesterCurriculum.swift:82]: add one more line at very last

File: TAT/Networking/API/APIManager+SemesterCurriculumList.swift [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+      return "/api/curriculums"
+    }
+
+    var method: Moya.Method {
+      return .get
+    }
+
+    var headers: [String: String]? {
+      guard let token = UserDefaults.standard.string(forKey: "token") else {
+        print("QAQ")
```
Comment (freddy5566) [inline TAT/Networking/API/APIManager+SemesterCurriculumList.swift]: same thing like above

**Comment by freddy5566** [comment]
I've rebased this branch onto develop, @kamisakihideyoshi  before starting develop new feature, please ` git pull --rebase ` to avoid issue, and Indent should use 2 spaces rather than tabs to conserve space and help prevent line wrapping. 
.ref: https://github.com/raywenderlich/swift-style-guide

**Pull request merged**